core/src/main/java/io/grpc/transport/ServerListener.java

```diff
@@ -44,5 +43,12 @@
    * @param transport the new transport to be observed.
    * @return a listener for stream creation events on the transport.
    */
-  ServerTransportListener transportCreated(Service transport);
+  ServerTransportListener transportCreated(ServerTransport transport);
+
+  /**
+   * The server is shutting down. No new transports will be processed, but existing streams may
+   * continue. Shutdown is only caused by a call to {@link Server#shutdown()}. All resources have
+   * been released.
+   */
+  void serverShutdown();
```

Comment:
Guava's Service had a single State, which I want to avoid because they were very painful to use. Entering TERMINATED means you are no longer STOPPING, and so if you want to do the check "can any new requests occur" you have to check for both TERMINATED and STOPPING _and_ you have to do in a non-racy way. It also means you deal with lots of different word forms: stopAsync() makes the state STOPPING, notifyStopped() makes the state TERMINATED.

But in any case, I don't want these words to differ from the ones in ClientTransport, ClientTransport.Listener, ChannelImpl, and ServerImpl. If I change them here, I feel I have to change them everywhere. In addition I don't have a good alternative to "shutdown" in {Channel,Server}Impl.isShutdown(). isStopping doesn't work because that isn't a permanent state. Using "stopped" instead of "shutdown" doesn't seem to solve the naming issue.